tests/scripts/all.sh

```diff
@@ -2780,6 +2780,104 @@ component_test_psa_crypto_config_reference_ecc_no_bignum () {
     #tests/ssl-opt.sh
 }
 
+# Helper for setting common configurations between:
+# - component_test_tfm_config_p256m_driver_accel_ec()
+# - component_test_tfm_config()
+common_tfm_config () {
+    # Enable TF-M config
+    cp configs/tfm_mbedcrypto_config_profile_medium.h "$CONFIG_H"
+    cp configs/crypto_config_profile_medium.h "$CRYPTO_CONFIG_H"
+
+    # Adjust for the fact that we're building outside the TF-M environment.
+    #
+    # TF-M has separation, our build doesn't
+    scripts/config.py unset MBEDTLS_PSA_CRYPTO_SPM
```

Comment:
Having just gotten bitten by this, I don't like the fact that the configuration in `configs` can't be built. The problem is that we can't do a build with `MBEDTLS_PSA_CRYPTO_SPM` without some third-party files — the library build succeeds but the tests fail:
```
In file included from tests/include/test/psa_crypto_helpers.h:27,
                 from tests/src/psa_crypto_helpers.c:26:
tests/include/test/psa_helpers.h:25:10: fatal error: spm/psa_defs.h: No such file or directory
   25 | #include "spm/psa_defs.h"
      |          ^~~~~~~~~~~~~~~~
compilation terminated.
```

It's also weird that we're not using `test-ref-configs` like we do for all the other configurations in the `configs` directory.

I think the files need to be reorganized. The `configs` directory should contain files that work on their own. Proposal:

* Move `tfm_mbedcrypto_config_profile_medium.h` and `crypto_config_profile_medium.h` to a subdirectory (and perhaps rename them). Note that we don't want to change these files because they're copies of the files from TF-M.
* Create a `configs/config-tfm.h` that:
    1. includes the TF-M mbedtls config file;
    2. sets `MBEDTLS_PSA_CRYPTO_CONFIG_FILE` to point to the TF-M psa crypto config file;
    3. unsets `MBEDTLS_PSA_CRYPTO_SPM`, and whatever else is needed to build .
* Add `configs/config-tfm.h` to `test-ref-configs.pl`.
* Create a file under `tests/configs` that arranges the exact TF-M configuration that we're using for code size measurements. This one doesn't need to produce working executables.

Reply:
Agreed, these files should not be in that directory. I'd rather not have a wrapper for them, but I do want to change how we handle them, so that the original content is undisturbed at the top and our modifications are clearly separate - the TF-M config files need to change, and having a wrapper to "set/unset ... whatever else is needed to build" is such a tight coupling between the two files in that proposed split I'd rather not have it in two files

Reply:
@yanrayw ^

Reply:
Marking as resolved by @yanrayw's PR - thanks!